### ISSUE TITLE
Error stickyScrollings[key].updateSize is not a function appears when used with Visualforce remoting in Salesforce

### ISSUE THREAD

**Issue by datapharmer** [fullcalendar/fullcalendar#5601]
When using main.js in a visualforce page in Salesforce, if the apex controller includes @remoteAction then it adds headers for VFRemote.js which causes an error with stickyScrollingss[key].updateSize

This results in an inability to move events on the calendar via drag/drop.

The problem did not exist with the most recent release of 4.x

JSFiddle example: https://jsfiddle.net/70o6s2vt/

Dev Console Errors:

Uncaught TypeError: stickyScrollings[key].updateSize is not a function
    at ScrollGrid.updateStickyScrolling (?editor_console=true:sourcemap:17679)
    at ScrollGrid.<anonymous> (?editor_console=true:sourcemap:17404)
    at ?editor_console=true:sourcemap:2766
    at Array.some (<anonymous>)
    at ?editor_console=true:sourcemap:2766
    at Array.some (<anonymous>)
    at T (?editor_console=true:sourcemap:2766)
    at ?editor_console=true:sourcemap:2766
    at Array.some (<anonymous>)
    at _ (?editor_console=true:sourcemap:2766)
    at flushToDom (?editor_console=true:sourcemap:2789)
    at DelayedRunner.Calendar._this.handleRenderRequest [as drainedOption] (?editor_console=true:sourcemap:11839)
    at DelayedRunner.drained (?editor_console=true:sourcemap:9693)
    at DelayedRunner.tryDrain (?editor_console=true:sourcemap:9675)
    at DelayedRunner.request (?editor_console=true:sourcemap:9637)
    at Calendar.render (?editor_console=true:sourcemap:11864)
    at HTMLDocument.<anonymous> (?editor_console=true:1868)
ScrollGrid.updateStickyScrolling @ ?editor_console=true:sourcemap:17679
(anonymous) @ ?editor_console=true:sourcemap:17404
(anonymous) @ ?editor_console=true:sourcemap:2766
(anonymous) @ ?editor_console=true:sourcemap:2766
T @ ?editor_console=true:sourcemap:2766
(anonymous) @ ?editor_console=true:sourcemap:2766
_ @ ?editor_console=true:sourcemap:2766
flushToDom @ ?editor_console=true:sourcemap:2789
Calendar._this.handleRenderRequest @ ?editor_console=true:sourcemap:11839
DelayedRunner.drained @ ?editor_console=true:sourcemap:9693
DelayedRunner.tryDrain @ ?editor_console=true:sourcemap:9675
DelayedRunner.request @ ?editor_console=true:sourcemap:9637
Calendar.render @ ?editor_console=true:sourcemap:11864
(anonymous) @ ?editor_console=true:1868

Uncaught (in promise) TypeError: stickyScrollings[key].updateSize is not a function
    at ScrollGrid.updateStickyScrolling (?editor_console=true:sourcemap:17679)
    at ScrollGrid.<anonymous> (?editor_console=true:sourcemap:17404)
    at ?editor_console=true:sourcemap:2766
    at Array.some (<anonymous>)
    at ?editor_console=true:sourcemap:2766
    at Array.some (<anonymous>)
    at T (?editor_console=true:sourcemap:2766)
    at ?editor_console=true:sourcemap:2766
    at Array.some (<anonymous>)
    at _ (?editor_console=true:sourcemap:2766)

**Comment by acerix** [comment]
Thanks for the demo, I get the same error.

It's hard to tell where the error is coming from though, best if you can provide a minimal demo:

https://fullcalendar.io/reduced-test-cases

There's also an existing issue about issues with the Salesforce environment which may cover this:

https://github.com/fullcalendar/fullcalendar/issues/4990
↑
Please visit the above issue, **give a** 👍  **to vote** ([?](https://github.com/blog/2119-add-reactions-to-pull-requests-issues-and-comments)), press the 🔈**Subscribe** button to receive updates.

**Comment by datapharmer** [comment]
Hi @acerix 

This is in visualforce/apex which is handled differently than lightning web components, and I can confirm that the workaround posted there does not work here.

That said, I dug in more and determined I can restore dragging capability by changing the line
`stickyScrollings[key].updateSize();`
to
```
if(typeof stickyScrollings[key].updateSize == 'function') {
	stickyScrollings[key].updateSize();}
```
This seems to still run the function on initialization and resize of the window, so I don't think it breaks anything - can someone more familiar with the code confirm this check is reasonable/doesn't hurt anything, and if so can we consider doing a commit to the repository?

**Comment by mbernwieser** [comment]
Experiencing the same error in an ember.js project. Can confirm that https://github.com/fullcalendar/fullcalendar-scheduler/pull/567 solves the issue.

**Comment by pixel9** [comment]
I'm seeing this exact issue using Fullcalendar scheduler 5.3.0 in a Salesforce VisualForce environment using remoting. The error is caused by VisualForce modifying the `Array` prototype.

https://github.com/fullcalendar/fullcalendar-scheduler/blob/7f0aec653338d0041e289ca075180c4d7ce04b0a/scrollgrid/src/ScrollGrid.tsx#L521

This line here doesn't verify the key using `hasOwnProperty` or something similar so it breaks when it loops over the array using `for...in` and an extra `remove` function that was monkeypatched in the Salesforce environment gets included.

This PR _would_ fix the symptom, 
https://github.com/fullcalendar/fullcalendar-scheduler/pull/567

...but the intent of the code would be more correct to iterate the array using a more traditional loop using an index. Something like this would do the trick:

```js
...
    for (let i = 0; i < stickyScrollings.length; i++) {
      stickyScrollings[i].updateSize()
    }
...

```

**Comment by negberts** [comment]
We are using angular and are experiencing the same issue when initializing the timeline grid without any further options. Is there any workaround that we can apply for now?

**Comment by arshaw** [comment]
I merged https://github.com/fullcalendar/fullcalendar-scheduler/pull/568 and will release it very soon

**Comment by arshaw** [comment]
I'll release this soon. see https://github.com/fullcalendar/fullcalendar/issues/5601#issuecomment-686858185

**Comment by arshaw** [comment]
this has been released in v5.3.1!
https://github.com/fullcalendar/fullcalendar/releases/tag/v5.3.1
